### PR TITLE
Fix console logging on sftp

### PR DESCRIPTION
**Pull request by shubhamgogna**
Updated the remote functions to return maps that can have an error string. #33 
The wrapped callbacks output to the JS console.

@joshcurtis @wmedrano

### CODE REVIEW

File: src/clj/server/client_interop.clj [hunk 1]
```diff
@@ -61,63 +61,75 @@
...
-    (format-ls (ssh/sftp (:sftp-chan @state/connection-state)
-                         {}
-                         :ls (if (string/blank? path) "./" path)))))
+  {:pre [(string? path)]}
+  (try
+    (let [non-blank-path (if (string/blank? path) "./" path)]
+      (locking state/ssh-lock
+        {:out (format-ls (ssh/sftp (:sftp-chan @state/connection-state) {} :ls non-blank-path))
+         :error nil}))
+  (catch Exception e {:out nil :error (.getMessage e)})))
```
Comment (joshcurtis) [inline src/clj/server/client_interop.clj:122]: The message of the error includes the entire stack trace. I find it's a bit spamming. 

Could we return something like "ls 'file path'  failed plus the error id?

File: src/clj/server/client_interop.clj [hunk 2]
```diff
@@ -61,63 +61,75 @@
...
 
 (defremote ssh-disconnect!
+  "Disconnects from SSH. Returns nil if successful, otherwise error message."
   []
-  (state/ssh-disconnect!)
-  nil)
+  (try
+    (state/ssh-disconnect!)
+    nil
+  (catch Exception e (.getMessage e))))
```
Comment (joshcurtis) [inline src/clj/server/client_interop.clj:69]: See comment about ls

.getMessage isn't useful IMO

File: src/clj/server/client_interop.clj (hunk 1 above)
Comment (joshcurtis) [inline src/clj/server/client_interop.clj:122]: This comment applies for all the commands.

**Comment by shubhamgogna** [comment]
@joshcurtis Can you pull and try it now? The stacktraces should be gone.

**Comment by joshcurtis** [comment]
I'm away from my computer. I will in a couple hours.

**Comment by joshcurtis** [comment]
I don't see any java stack traces. I'm happy.

**Pull request merged**